lib/cmd_tool_is4In6.go

```diff
@@ -0,0 +1,54 @@
+package lib
+
+import (
+	"fmt"
+	"github.com/spf13/pflag"
+)
+
+// CmdToolIs4In6Flags are flags expected by CmdToolIs4In6
+type CmdToolIs4In6Flags struct {
+	Help bool
+}
+
+// Init initializes the common flags available to CmdToolIs4In6 with sensible
+func (f *CmdToolIs4In6Flags) Init() {
+	pflag.BoolVarP(
+		&f.Help,
+		"help", "h", false,
+		"show help.",
+	)
+}
+
+// CmdToolIs4In6 checks if given ip is an IPv4-mapped IPv6 address.
+func CmdToolIs4In6(f CmdToolIs4In6Flags, args []string, printHelp func()) error {
+	if f.Help {
+		printHelp()
+		return nil
+	}
+
+	op := func(input string, inputType INPUT_TYPE) error {
+		switch inputType {
+		case INPUT_TYPE_IP:
+			fmt.Printf("%s %t\n", input, is4in6(input))
+		default:
+			return ErrInvalidInput
+		}
+		return nil
+	}
+
+	return GetInputFrom(args, true, true, op)
+}
+
+func is4in6(ip string) bool {
```

Comment:
Is there no func for this within the netip library we intended on using for all of these?

Reply:
yes. net library doesn't have a function implementing this. It is defined in netIP library ( which we are using as inspiration ) but its not imported